game-core/src/main/java/games/strategy/triplea/delegate/FinishedBattle.java

```diff
@@ -67,8 +67,8 @@ public Change addAttackChange(final Route route, final Collection<Unit> units,
     }
     final Territory attackingFrom = route.getTerritoryBeforeEnd();
     attackingUnits.addAll(units);
-    m_attackingFromMap.putIfAbsent(attackingFrom, new ArrayList<>());
-    final Collection<Unit> attackingFromMapUnits = m_attackingFromMap.get(attackingFrom);
+    final Collection<Unit> attackingFromMapUnits = m_attackingFromMap.computeIfAbsent(attackingFrom,
+        i -> new ArrayList<>());
```

Comment:
I'm okay with it, but that's probably because I've been reading and writing code like this for a few years now, mostly in the context of a `ConcurrentMap`, where performing these multiple operations atomically was required, not just because it was less code. :smile:  It may just be one of those things that you don't think twice about once you've done it enough.